nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/engine/StatelessFlowManager.java

```diff
@@ -270,10 +271,9 @@ public FlowFileQueue createFlowFileQueue(final LoadBalanceStrategy loadBalanceSt
     public ReportingTaskNode createReportingTask(final String type, final String id, final BundleCoordinate bundleCoordinate, final Set<URL> additionalUrls, final boolean firstTimeAdded,
                                                  final boolean register, final String classloaderIsolationKey) {
 
-        if (type == null || id == null || bundleCoordinate == null) {
```

Comment:
With this removal will the log/exception information still be as specific as it was?  The code is clearly cleaner but we go backwards in communicating intent.  Am i reading that right?

Reply:
Valid point. I used this implementation because it was cleaner as well. The requireNonNull throws a generic NullPointerException, but doesn't have the full details like the current code:
throw new NullPointerException("Must supply type, id, and bundle coordinate in order to create Reporting Task. Provided arguments were type=" + type + ", id=" + id + ", bundle coordinate = " + bundleCoordinate);
                
I can change this back if desired, just let me know.

Reply:
Yeah I'd not change this since while the code is cleaner it is also far less effective in conveying intent to the user.  Ultimately that is our desire given this block of code.

Reply:
Done :)